### PR TITLE
Wall rendering

### PR DESCRIPTION
**Pull request by tomc1998**

### CODE REVIEW

File: src/engine/comp/wall.hpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 #include "engine/entity_id.hpp"
 #include "engine/vec.hpp"
+#include "engine/resource.hpp"
```
Comment (Pomchee2) [inline src/engine/comp/wall.hpp]: include texture.hpp instead

File: src/engine/comp/wall.hpp [hunk 2]
```diff
@@ -9,8 +10,9 @@
 
 class CompWall {
 public:
-  CompWall(EntityId entity_id, std::vector<Vec2> vertices);
+  CompWall(EntityId entity_id, std::vector<Vec2> vertices, ResHandle edge_tex);
```
Comment (Pomchee2) [inline src/engine/comp/wall.hpp]: Change to TexHandle

File: src/engine/comp/wall.hpp [hunk 3]
```diff
@@ -9,8 +10,9 @@
...
 
   EntityId entity_id;
   std::vector<Vec2> vertices;
+  ResHandle edge_tex;
```
Comment (Pomchee2) [inline src/engine/comp/wall.hpp]: ditto

File: src/engine/renderer/paint_controller.cpp [hunk 4]
```diff
@@ -36,3 +36,22 @@ void PaintController::fill_rect(f32 x, f32 y, f32 w, f32 h, Color *color) {
                 Vertex(Vec2(x + w, y + h), color, Vec2(uvs[2], uvs[3]))};
   curr_batch.buffer(v, 6);
 }
+
+void PaintController::draw_quads(Vertex *v_buf, size_t num_quads,
+                                 ResHandle tex) {
+  flush_if_batch_tex_not(res_manager->lookup_tex(tex)->cache_tex_ix);
```
Comment (Pomchee2) [inline src/engine/renderer/paint_controller.cpp]: Why not make this use the get_tex_for_handle method?

File: src/test_screen.cpp [hunk 5]
```diff
@@ -5,22 +5,26 @@
 #include "engine/screen.hpp"
 #include "engine/vec.hpp"
 #include "test_screen.hpp"
+#include "engine/resource_manager.hpp"
+#include "engine/resource.hpp"
```
Comment (Pomchee2) [inline src/test_screen.cpp]: Change to texture.hpp

File: src/engine/comp/wall.hpp (hunk 1 above)
Comment (tomc1998) [reply src/engine/comp/wall.hpp]: This isn't merged yet

File: src/engine/comp/wall.hpp (hunk 2 above)
Comment (tomc1998) [reply src/engine/comp/wall.hpp]: Not merged yet

File: src/test_screen.cpp (hunk 5 above)
Comment (tomc1998) [reply src/test_screen.cpp]: Not merged yet

**Pull request merged**